include/tscore/MT_hashtable.h

```diff
@@ -295,6 +353,37 @@ IMTHashTable<key_t, data_t>::first_entry(int bucket_id, HashTableIteratorState<k
   return (data_t)0;
 }
 
+template <class key_t, class data_t>
+inline data_t
+IMTHashTable<key_t, data_t>::first_entry_by_key(int bucket_id, key_t key, HashTableIteratorState<key_t, data_t> *s)
```

Comment:
From the previous comments, this should just be `find` and it should return an iterator to the first element with that key. The iterator state should be embedded in that class, rather than explicitly passed.